src/containers/Tenant/Diagnostics/TopicData/TopicData.tsx

```diff
@@ -0,0 +1,165 @@
+            setStartOffset(convertToNumber(selectedPartitionData.startOffset));
+            setEndOffset(convertToNumber(selectedPartitionData.endOffset));
+        }
+    }, [selectedPartition, partitions]);
+
+    const numericSelectedOffset = React.useMemo(() => {
+        return convertToNumber(selectedOffset);
+    }, [selectedOffset]);
+    const numericStartTimestamp = React.useMemo(() => {
+        return convertToNumber(startTimestamp);
+    }, [startTimestamp]);
+
+    const tableFilters = React.useMemo(() => {
+        return {
+            path,
+            database,
+            partition: selectedPartition ?? '',
+            selectedOffset: numericSelectedOffset,
+            startTimestamp: numericStartTimestamp,
+            topicDataFilter,
+        };
+    }, [
+        path,
+        database,
+        selectedPartition,
+        numericSelectedOffset,
+        numericStartTimestamp,
+        topicDataFilter,
+    ]);
+
+    const {columnsToShow, columnsToSelect, setColumns} = useSelectedColumns(
+        getAllColumns(setFullValue),
+        TOPIC_DATA_SELECTED_COLUMNS_LS_KEY,
+        TOPIC_DATA_COLUMNS_TITLES,
+        DEFAULT_TOPIC_DATA_COLUMNS,
+        REQUIRED_TOPIC_DATA_COLUMNS,
+    );
+
+    const renderControls: RenderControls = () => {
+        return (
+            <TopicDataControls
+                // component has uncontrolled components inside, so it should be rerendered on filters reset
+                key={controlsKey}
+                columnsToSelect={columnsToSelect}
+                handleSelectedColumnsUpdate={setColumns}
+                partitions={partitions}
+                partitionsLoading={partitionsLoading}
+                partitionsError={partitionsError}
+                initialOffset={startOffset}
+                endOffset={endOffset}
+            />
+        );
+    };
+
+    const renderEmptyDataMessage = () => {
+        const hasFilters = selectedOffset || startTimestamp;
+
+        const resetFilter = () => {
+            dispatch(setSelectedOffset(undefined));
+            dispatch(setStartTimestamp(undefined));
+            setControlsKey((prev) => prev + 1);
+        };
+
+        return (
+            <EmptyFilter
+                title={i18n('label_nothing-found')}
+                message={i18n('description_nothing-found')}
+                onShowAll={hasFilters ? resetFilter : undefined}
+                showAll={i18n('action_show-all')}
+                image={<NoSearchResults width={230} height={230} />}
+            />
+        );
+    };
+
+    return (
+        <React.Fragment>
+            <FullValue value={fullValue} onClose={() => setFullValue(undefined)} />
+            <ResizeablePaginatedTable
+                //if any filter change, we need to initiate a new table component to ensure correct start offset calculations in generateTopicDataGetter
+                key={topicDataFilter + selectedOffset + startTimestamp + selectedPartition}
```

Comment:
Table chunks are reset every time value in `filters` changes, why do you need key here?

Reply:
Because of logic inside `generateTopicDataGetter` https://github.com/ydb-platform/ydb-embedded-ui/pull/2145/files/c38c503b6fcc6c99e35da4324056d8374ccb5c03#diff-55d62e69373cb3d825abe22e33d121b8cda18477596ea4d1038a5560934ac1f4 ! Every time we should calculate initial offset, and table offset should be reset to `0`.

Reply:
But there is already 

 parentRef.current.scrollTo(0, 0);

in  React.useLayoutEffect in PaginatedTable

shouldnt this be enough?

Reply:
I made a little improvement to PaginatedTable, so it does't refetch previous active chunks when filters changes. So the key is not needed any more.